ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java

```diff
@@ -248,22 +254,43 @@ private boolean updateBlobMetadata(BlobId blobId, String fieldName, Object value
     Objects.requireNonNull(fieldName, "Field name cannot be null");
 
     // We update the blob metadata value in two places:
-    // 1) the CosmosDB metadata collection
-    // 2) the blob storage entry metadata (to enable rebuilding the database)
-
+    // 1) the blob storage entry metadata (so GET's can be served entirely from ABS)
+    // 2) the CosmosDB metadata collection
     try {
-      if (!azureBlobDataAccessor.updateBlobMetadata(blobId, fieldName, value)) {
-        // TODO: what if this is a retry where ABS has been updated but Cosmos has not?
-        return false;
+      boolean updatedStorage, updatedCosmos = false;
```

Comment:
nit: could you separate these two assignments onto separate lines? it looks a little confusing since one of the values is assigned/mutated but the other is not assigned until line 265